prov/sockets/src/sock_av.c

```diff
@@ -527,7 +527,7 @@ int sock_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 
 		_av->table_hdr = mmap(NULL, table_sz, PROT_READ | PROT_WRITE,
 					MAP_SHARED, _av->shared_fd, 0);
-		_av->attr.map_addr = _av->table_hdr + sizeof(_av->table_hdr);
+		_av->attr.map_addr = _av->table_hdr + sizeof(struct sock_av_table_hdr);
```

Comment:
This looks wrong.  See the coverity report:

suspicious_pointer_arithmetic: Adding 8UL /* sizeof (_av->table_hdr) */ to pointer _av->table_hdr of type struct sock_av_table_hdr * is suspicious because adding an integral value to this pointer automatically scales that value by the size, 24 bytes, of the pointed-to type, struct sock_av_table_hdr. Most likely, sizeof (_av->table_hdr) is extraneous and should be replaced with 1.

It definitely seems that the change should be to add 1, not a sizeof(anything).  That would only work if table_hdr were a void *.